packages/client/src/views/RecordAudit/ActionMenu.tsx

```diff
@@ -0,0 +1,646 @@
+          dispatch(
+            goToCertificateCorrection(
+              declarationId as string,
+              CorrectionSection.Corrector
+            )
+          )
+        }}
+        disabled={!isDownloaded}
+      >
+        <Icon name="NotePencil" color="currentColor" size="large" />
+        {intl.formatMessage(messages.correctRecord)}
+      </DropdownMenu.Item>
+    )
+  )
+}
+
+const ArchiveAction: React.FC<
+  IActionItemCommonProps & { toggleDisplayDialog?: () => void }
+> = ({ toggleDisplayDialog, intl, isDownloaded, declarationStatus, scope }) => {
+  const isArchivable =
+    declarationStatus &&
+    [
+      SUBMISSION_STATUS.IN_PROGRESS,
+      SUBMISSION_STATUS.DECLARED,
+      SUBMISSION_STATUS.VALIDATED,
+      SUBMISSION_STATUS.REJECTED
+    ].includes(declarationStatus as SUBMISSION_STATUS)
+
+  // @ToDo use: `record.registration-archive` after configurable role pr is merged
+  // @Question: If user has archive scope but not register scope,
+  // can he archive validated record?
+  const userHasArchiveScope =
+    scope &&
+    ((scope as any as string[]).includes('register') ||
+      ((scope as any as string[]).includes('validate') &&
+        declarationStatus !== SUBMISSION_STATUS.VALIDATED))
+
+  return (
+    isArchivable &&
+    userHasArchiveScope && (
+      <DropdownMenu.Item onClick={toggleDisplayDialog} disabled={!isDownloaded}>
+        <Icon name="Archive" color="currentColor" size="large" />
+        {intl.formatMessage(messages.archiveRecord)}
+      </DropdownMenu.Item>
+    )
+  )
+}
+
+const ReinstateAction: React.FC<
+  IActionItemCommonProps & { toggleDisplayDialog?: () => void }
+> = ({ toggleDisplayDialog, isDownloaded, intl, declarationStatus, scope }) => {
+  const isArchived = declarationStatus === SUBMISSION_STATUS.ARCHIVED
+
+  // @ToDo use: `record.registration-reinstate` after configurable role pr is merged
+  // @Question: If user has reinstate scope but not register scope,
+  // can he reinstate validated record?
+  const userHasReinstateScope =
+    scope &&
+    ((scope as any as string[]).includes('register') ||
+      (scope as any as string[]).includes('validate'))
+  return (
+    isArchived &&
+    userHasReinstateScope && (
+      <DropdownMenu.Item onClick={toggleDisplayDialog} disabled={!isDownloaded}>
+        <Icon name="FileArrowUp" color="currentColor" size="large" />
+        {intl.formatMessage(messages.reinstateRecord)}
+      </DropdownMenu.Item>
+    )
+  )
+}
+
+const ReviewAction: React.FC<
+  IActionItemCommonProps &
+    IDeclarationProps & { isDuplicate: boolean; goToPage: typeof goToPage }
+> = ({
+  declarationId,
+  declarationStatus,
+  type,
+  scope,
+  isDownloaded,
+  isDuplicate,
+  intl,
+  goToPage
+}) => {
+  const isPendingCorrection =
+    declarationStatus === EVENT_STATUS.CORRECTION_REQUESTED
+
+  const isReviewableDeclaration =
+    declarationStatus &&
+    [EVENT_STATUS.DECLARED, EVENT_STATUS.VALIDATED].includes(declarationStatus)
```

Comment:
@Zangetsu101 these `isFoo` logic stuff, could/should we move them to a centralized place near `useDeclaration`? I'm a bit worried that stuff like
```
  const isArchivable =
    declarationStatus &&
    [
      SUBMISSION_STATUS.IN_PROGRESS,
      SUBMISSION_STATUS.DECLARED,
      SUBMISSION_STATUS.VALIDATED,
      SUBMISSION_STATUS.REJECTED
    ].includes(declarationStatus as SUBMISSION_STATUS)
```
get defined in multiple places in the application and they start to mismatch and get outdated in the future.

The scope related stuff should lay in the `usePermissions` but that requires the scopes to be merged I think.

Reply:
We'll probably have to make some changes to this after the scope PR's merged in